test/unit/org/apache/cassandra/cql3/RandomSchemaV2Test.java

```diff
@@ -0,0 +1,326 @@
+import org.apache.cassandra.cql3.ast.TableReference;
+import org.apache.cassandra.cql3.ast.Txn;
+import org.apache.cassandra.db.Keyspace;
+import org.apache.cassandra.db.marshal.AbstractType;
+import org.apache.cassandra.db.marshal.CompositeType;
+import org.apache.cassandra.db.marshal.DurationType;
+import org.apache.cassandra.db.marshal.MapType;
+import org.apache.cassandra.db.marshal.UTF8Type;
+import org.apache.cassandra.index.Index;
+import org.apache.cassandra.index.SecondaryIndexManager;
+import org.apache.cassandra.index.sai.StorageAttachedIndex;
+import org.apache.cassandra.schema.ColumnMetadata;
+import org.apache.cassandra.schema.KeyspaceMetadata;
+import org.apache.cassandra.schema.Schema;
+import org.apache.cassandra.schema.TableMetadata;
+import org.apache.cassandra.service.consensus.TransactionalMode;
+import org.apache.cassandra.transport.ProtocolVersion;
+import org.apache.cassandra.utils.AbstractTypeGenerators;
+import org.apache.cassandra.utils.CassandraGenerators;
+import org.assertj.core.api.Assertions;
+import org.awaitility.Awaitility;
+
+import static accord.utils.Property.qt;
+import static java.lang.String.format;
+
+public class RandomSchemaV2Test extends CQLTester
+{
+    @BeforeClass
+    public static void setUpClass()
+    {
+        prePrepareServer();
+
+        // When values are large SAI will drop them... soooo... disable that... this test does not care about perf but correctness
+        DatabaseDescriptor.getRawConfig().sai_frozen_term_size_warn_threshold = null;
+        DatabaseDescriptor.getRawConfig().sai_frozen_term_size_fail_threshold = null;
+
+        // Once per-JVM is enough
+        prepareServer();
+    }
+
+    enum Mode { AccordEnabled, Default}
+
+    @Test
+    public void test()
+    {
+        requireNetwork();
+        Gen<ProtocolVersion> clientVersionGen = rs -> {
+            if (rs.decide(.8)) return null;
+            return rs.pickOrderedSet(ProtocolVersion.SUPPORTED);
+        };
+        Gen<Mode> modeGen = Gens.enums().all(Mode.class);
+        qt().withExamples(100).forAll(Gens.random(), modeGen, clientVersionGen).check((rs, mode, protocolVersion) -> {
+            clearState();
+
+            KeyspaceMetadata ks = createKeyspace(rs);
+            TableMetadata metadata = createTable(rs, ks.name);
+            if (mode == Mode.AccordEnabled)
+            {
+                // enable accord
+                schemaChange(String.format("ALTER TABLE %s WITH " + TransactionalMode.full.asCqlParam(), metadata));
+                metadata = Objects.requireNonNull(Schema.instance.getTableMetadata(ks.name, metadata.name));
+            }
+            Map<ColumnMetadata, CreateIndexDDL> indexedColumns = createIndex(rs, metadata);
+            Mutation mutation = nonTransactionMutation(rs, metadata);
+            if (mode == Mode.AccordEnabled)
+                mutation = mutation.withoutTimestamp().withoutTTL(); // Accord doesn't allow custom timestamps or TTL
```

Comment:
we don't actually block `TTL` atm... I have a patch for that... I have too many things open atm...